tools/ami-cleanup/src/internal/aws.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO move this under /libs at some point, however it is nowhere near
+// large enough to justify separate module today
+package internal
```

Comment:
`internal` is not a very good package name - it doesn't say anything about what the code in the package does.

https://go.dev/blog/package-names

> Good package names make code better. A package’s name provides context for its contents, making it easier for clients to understand what the package is for and how to use it. The name also helps package maintainers determine what does and does not belong in the package as it evolves. Well-named packages make it easier to find the code you need.

Reply:
This decision was based on the Go standard project layout [here](https://github.com/golang-standards/project-layout/tree/master/internal), which states that for smaller projects extra structure is not required, and that this code can live under a package named `internal`. [One of the examples](https://github.com/influxdata/influxdb/tree/master/internal) provided in the standard uses this naming scheme.

Do you want me to change this? If so, how do you want me to structure the project?

Reply:
Yeah, I remember when that repo was published there was a whole buzz about it because it is not in any way a "Go standard project layout." This is random internet people who slapped the term "standard" on their project.

I'd be much more inclined to adopt the device in the official Go blog (linked above).

In any case, the point of internal packages is that they reside in the subtree of a directory named internal, not that the package itself is necessarily named internal. This prevents them from being imported outside of the module but still gives you the opportunity to select a package name that better describes its contents.